pybossa/exporter/csv_reports_export.py

```diff
@@ -75,25 +75,23 @@ def _respond_csv(self, ty, project_id, info_only=False, **kwargs):
 
     def _make_zip(self, project, ty, **kwargs):
         name = self._project_name_latin_encoded(project)
-        csv_task_generator = self._respond_csv(ty, project.id)
-        if csv_task_generator is not None:
-            datafile = tempfile.NamedTemporaryFile()
-            try:
-                datafile.write(str(csv_task_generator).encode('utf-8'))
+        csv_task_str = self._respond_csv(ty, project.id, **kwargs)
+        if csv_task_str is not None:
+            with tempfile.NamedTemporaryFile() as datafile, \
+                 tempfile.NamedTemporaryFile(delete=False) as zipped_datafile:
+                datafile.write(csv_task_str.encode())
                 datafile.flush()
```

Comment:
1. The if block could be simplified
2. zipped_datafile can be created later as it was under original code as its unused under with block
3. With no data under csv (csv_task_str `None`, code not returning dict)

```python
if not csv_task_str:
     return {}
with tempfile.NamedTemporaryFile() as datafile:
     datafile.write(csv_task_str.encode())
     datafile.flush()
zipped_datafile = tempfile.NamedTemporaryFile(delete=False)
...
```